posts/benchmarking_design_decisions/index.qmd

```diff
@@ -0,0 +1,246 @@
+---
+title: "Using benchmarking to guide the adoption of dependencies in R packages"
+author:
+  - name: "James Mba Azam"
+    orcid: "https://orcid.org/0000-0001-5782-7330"
+date: last-modified
+categories: [benchmarking, R, tidyverse, cli, R packages, software architecture]
+format:
+  html:
+    toc: true
+execute: 
+  cache: true
+---
+
+```{r, echo = FALSE}
+knitr::opts_chunk$set(
+  cache = TRUE,
+  message = FALSE
+)
+```
+
+R package developers are constantly faced with design decisions about user
+friendliness. Such design decisions often include checking that inputs are
+rightly specified and returning feedback early, deciding on how to deal with
+issues arising during code execution within a function, and how to
+structure outputs.
+
+These design decisions often result in taking on dependencies. Dependencies are
```

Comment:
Not sure I agree with this definition of what a dependency is. A function could work properly without calling code from an external package, but it might be better to not rewrite that code, or the dependency could provide a more feature-rich function than base R. I would go with the more general definition used here: https://r-pkgs.org/dependencies-mindset-background.html

Reply:
Thanks. I've revised it here d08b3d6254cbb31a44d4e38ad2fd42c617d0bd80.